server/tserver/src/main/java/org/apache/accumulo/tserver/AssignmentHandler.java

```diff
@@ -154,6 +157,28 @@ public void run() {
         throw new RuntimeException("Minor compaction after recovery fails for " + extent);
       }
 
+      final Location expectedLocation = Location.future(this.server.getTabletSession());
+      try (Ample.ConditionalTabletsMutator mutator =
```

Comment:
Still need to take the code that sets the location that pull it here and combine it with this.  Following `TabletStateStore.setLocation` you through multiple layers of abstraction, will eventually find the impl that can be moved here.  Then all that code and and the layers could be removed.  This is the only place that calls TabletStateStore.setLocation